File: observer/observer_go/Current.go
package main

import "fmt"

// Current is an implement of Observer and Display,
type Current struct {
	temperature float64
	humidity    float64
	subject     Subject
}

func (c *Current) update() {
	c.temperature = c.subject.getTemperature()
	c.humidity = c.subject.getHumidity()
	c.display()
}

func (c *Current) display() {
	fmt.Printf("Current conditions: %f F degrees and %f humidity\n", c.temperature, c.humidity)
}

func (c *Current) setSubject(s Subject) {
	c.subject = s
}

File: observer/observer_ts/src/Current.ts
import { Observer } from "./Observer";
import { Display } from "./Display";
import { WeatherStation } from "./WeatherStation";

export class Current implements Observer, Display {
    private temperature: number
    private humidity: number
    constructor(private weatherData: WeatherStation) {
        this.temperature = 0
        this.humidity = 0
    }
    update() {
        this.temperature = this.weatherData.getTemperature()
        this.humidity = this.weatherData.getHumidity()
        this.display()
    }
    display() {
        console.log(`Current Conditions: ${this.temperature}F degrees and ${this.humidity}% humidity`);
    }
}